inf-clojure.el

```diff
@@ -843,6 +848,139 @@ HOST is the host the process is running on, PORT is where it's listening."
+  "Starts a socket REPL server and connects to it via `inf-clojure'.
+ARGS are optional, in the case where PORT is not chosen one is
+randomly selected between 5500 and 6000. The default for
+REPL-TYPE is clojure. If you are inside a project, `inf-clojure-socket' will always run in the project directory.
+
+ARGS must come in pairs ATTRIBUTE VALUE.
+
+The following attributes are recognized:
+
+`:host'
+
+VALUE must be a string specifying the address that the socket is using. By default it is `localhost'
+
+`:port'
+
+VALUE must be a number specifying the port for the socket to listen.
+
+`:repl-type'
+
+VALUE must be a valid repl type, given as a string e.g clojure/cljs/lein/babashka. Note that `joker' and `lumo' are not supported.
+
+`:project-name'
+
+VALUE can be any string, by default this is either derived from your current directory if you are inside a project or `standalone'
+
+`:opts'
+
+VALUE can be any string, by default this is nil and nothing is
+added to the startup form is chosen according to your
+`repl-type'. This is useful when you want to add some arguments
+or flags for your chosen tool. e.g `-Adev:test' for `clojure'"
+
+  (interactive)
+  (setq args (purecopy args)) ;; not sure if purecopy is useful, set-face-attribute does it.
+  (let* ((host (or (plist-get args :host) "localhost"))
+         (port (or (plist-get args :port) (+ 5500 (random 500))))
+         (repl-type (or (plist-get args :repl-type) "clojure"))
+         (opts (plist-get args :opts))
+         (project-name (or (plist-get args :project-name)
+                           (inf-clojure--project-name (or (clojure-project-dir)
+                                                          "standalone"))))
+         (socket-process-name (format "*%s-%s-socket-server*" project-name repl-type))
+         (socket-buffer-name (format "*%s-%s-socket*" project-name repl-type))
+         (socket-buffer (get-buffer-create socket-buffer-name))
+         (repl-buffer-name (format "%s-%s-repl" project-name repl-type))
+         (socket-form (cdr (assoc (intern repl-type) inf-clojure-socket-forms)))
+         (socket-cmd (apply 'concat
+                            (format socket-form (number-to-string port))
+                            (when opts `(" " ,opts))))
+         (sock (let ((default-directory (or (clojure-project-dir)
+                                            default-directory)))
+                 (start-file-process-shell-command
+                  socket-process-name socket-buffer
+                  socket-cmd))))
+    (with-current-buffer socket-buffer
+      (setq-local
+       inf-clojure-socket-callback
+       (lambda ()
+         (setq inf-clojure-custom-repl-type
+               (intern repl-type)
+               inf-clojure-custom-repl-name
+               ;; comint mode includes the ** chars
+               repl-buffer-name
+               repl-buffer
+               (get-buffer-create repl-buffer-name))
+         (inf-clojure-connect host port)
```

Comment:
This seems to be the only usage of `host`. So that means we always create the process on the local machine (that makes sense) but then we optionally try bind against a remote machine which seems wrong. If we accept a host, it seems like that information needs to be included when starting our socket repl? but I'm not clear what values host could be.

Reply:
host is now always "localhost", we're passing it in two places `inf-clojure-connect` and a `message` but not the actual socket startup as you pointed out - I think this makes sense now to keep as a let variable since it's not configurable.